clang/test/CheckedCRewriter/struct_init_list.c

```diff
@@ -0,0 +1,95 @@
+// RUN: cconv-standalone -alltypes %s -- | FileCheck -match-full-lines %s
+// RUN: cconv-standalone -alltypes %s -- | %clang_cc1 -fignore-checkedc-pointers -fno-builtin -verify -fcheckedc-extension -x c -
+// expected-no-diagnostics
+
+struct foo {
+  int (*fp)(int *p); 
+  // CHECK:  _Ptr<int (int *)> fp;
+};
```

Comment:
If you are assigning an external function, should the `fp` be `_Ptr` or just `(*)`? Does Checked C's `clang` compile this file properly? You could add that to the top of `RUN` to see.

Reply:
I thought the second RUN line would do that, but apparently not. I do in fact get an error when I compile, but not quite in the way you expect. Line 13 is changed to `f.fp = _Assume_bounds_cast<_Ptr<int (int *)> >(&xfunc);`, but the compiler doesn't like this. Removing the cast lets it compile without issue.

Reply:
This is due to a recent change by @Machiry. I'm not sure when it decides to add things. Maybe investigate that?

Reply:
Or: Maybe the better thing is to force it to be `int (*fp)(int *p)` and not a `_Ptr` ?